mgmake/generate.go

```diff
@@ -18,77 +21,132 @@ type templateTargets struct {
 	Args       []string
 }
 
-// GenerateMakefile is a mage target that ...
-func GenerateMakefile(makefile string) error {
-	const defaultTargets = "mage_targets"
-	genDir := filepath.Dir(makefile)
-	mgDefaultTargets := fmt.Sprintf("%s.mk", filepath.Join(genDir, strcase.ToKebab(defaultTargets)))
-	if makefile == mgDefaultTargets {
-		return fmt.Errorf("%s has the same name as the default %s makefile", makefile, mgDefaultTargets)
+// nolint: gochecknoglobals
+var (
+	defaultTarget string
+	makefiles     = make(map[string]Makefile)
+)
+
+type Makefile struct {
+	Namespace     interface{}
+	Path          string
+	DefaultTarget string
+}
+
+func AddExtraMakefiles(mks []Makefile) {
+	for _, i := range mks {
+		if i.Namespace == nil {
+			panic("Namespace needs to be defined")
+		}
+		if i.Path == "" {
+			panic("Path needs to be defined")
+		}
+		namespace := reflect.TypeOf(i.Namespace).Name()
+		makefiles[toMakeTarget(namespace)] = i
 	}
-	mglog.Logger("generate-makefile").Info("generating Makefile...")
+}
+
+const defaultNamespace = "default"
+
+// GenerateMakefiles is a mage target that ...
```

Comment:
> that ...

should we fill it out before merge mayhaps?

Reply:
Sure!